Feat: add support for automatic login

little-thing opened the pull request:
### What problem does this PR solve?

####  Newly added logic to automatically log in after the login expires.
In the /web/.env file, set LOGIN_TYPE to auto, and implement the autoLogin function in the authorization-util.ts file to enable automatic login. If the autoLogin function is not set, a warning window will pop up.
Automatic login supports returning to the original page after logging in.

####  Fixed the issue where the SSO login error message could not be displayed due to page redirection.

####  Fix the issue in chat-api-key-modal.tsx where only one token is allowed to be set.

####  auth-hooks.ts is used to obtain the user's login status; there should not be any logic for redirecting to a page here.

### Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

asiroliu commented:
@KevinHuSh 
LGTM

Verified on the following pages:
https://platform.openai.com/docs/models/gpt-4.1
https://platform.openai.com/docs/models/gpt-4.1-mini
https://platform.openai.com/docs/models/gpt-4.1-nano

little-thing commented:
@asiroliu   I updated the automatic login logic to match the new OAuth implementation and simplified the injection method for LOGIN_TYPE. Currently, you only need to set LOGIN_TYPE=auto in the .env file to enable automatic login.

KevinHuSh commented:
Appreciations!
By our evaluations, this feature is not going to be merged into main branch.
